spynnaker/pyNN/models/neuron/abstract_population_vertex.py

```diff
@@ -828,204 +890,174 @@ def reset_to_first_timestep(self):
-                spikes_per_second, 0, n_connections)
-            total_weights[synapse_type] += spikes_per_tick * (
-                weight_max * n_connections)
-
-            if synapse_dynamics.are_weights_signed():
-                weights_signed = True
-
-        max_weights = numpy.zeros(n_synapse_types)
-        for synapse_type in range(n_synapse_types):
-            if delay_running_totals[synapse_type].variance == 0.0:
-                max_weights[synapse_type] = max(total_weights[synapse_type],
-                                                biggest_weight[synapse_type])
-            else:
-                stats = running_totals[synapse_type]
-                rates = rate_stats[synapse_type]
-                max_weights[synapse_type] = min(
-                    self._ring_buffer_expected_upper_bound(
-                        stats.mean, stats.standard_deviation, rates.mean,
-                        stats.n_items, self.__ring_buffer_sigma),
-                    total_weights[synapse_type])
-                max_weights[synapse_type] = max(
-                    max_weights[synapse_type], biggest_weight[synapse_type])
-
-        # Convert these to powers; we could use int.bit_length() for this if
-        # they were integers, but they aren't...
-        max_weight_powers = (
-            0 if w <= 0 else int(math.ceil(max(0, math.log(w, 2))))
-            for w in max_weights)
-
-        # If 2^max_weight_power equals the max weight, we have to add another
-        # power, as range is 0 - (just under 2^max_weight_power)!
-        max_weight_powers = (
-            w + 1 if (2 ** w) <= a else w
-            for w, a in zip(max_weight_powers, max_weights))
-
-        # If we have synapse dynamics that uses signed weights,
-        # Add another bit of shift to prevent overflows
-        if weights_signed:
-            max_weight_powers = (m + 1 for m in max_weight_powers)
-
-        return list(max_weight_powers)
+            weight_max = synapse_dynamics.get_weight_maximum(
+                connector, synapse_info)
+            weight_max *= weight_scale
 
-    @staticmethod
-    def __get_weight_scale(ring_buffer_to_input_left_shift):
-        """ Return the amount to scale the weights by to convert them from \
-            floating point values to 16-bit fixed point numbers which can be \
-            shifted left by ring_buffer_to_input_left_shift to produce an\
-            s1615 fixed point number
+            weight_scale_limit = float(DataType.S1615.scale)
+            if weight_scale_limit * min_weights[synapse_type] < weight_max:
+                max_weight = self.__get_closest_weight(weight_max)
+                min_weights[synapse_type] = max_weight / weight_scale_limit
 
-        :param int ring_buffer_to_input_left_shift:
-        :rtype: float
+        self.__check_weights(min_weights, weight_scale, incoming_projections)
+
+        return min_weights
+
+    def __check_weights(
```

Comment:
This code does not appear to run on user supplied min_weights

Would it not be better if it did.

Reply:
Fixed this - now runs on user supplied values (via population additional_parameters).